packages/analytics/add-cloud-link-handler/index.ts

```diff
@@ -0,0 +1,52 @@
+import { getProductIntentFromURL } from '../get-product-intent-from-url'
+import { getUTMParamsCaptureState } from '../utm-params-capture'
+
+const destinations: string[] = [
+  'portal.cloud.hashicorp.com',
+  'app.terraform.io',
+]
+const containsDestination = (str: string): boolean =>
+  destinations.some(function (destination) {
+    return str.indexOf(destination) >= 0
+  })
+
+export function addCloudLinkHandler() {
+  if (typeof window === 'undefined') return
+
+  window.addEventListener('click', (event) => {
+    /**
+     * Current <Button /> component implementation wraps the
+     * inner button elements within a <span /> which requires us
+     * to traverse up the dom to the closest anchor tag.
+     * See {@link https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath}
+     */
+    const composedPath = event.composedPath()
+
+    composedPath.forEach((path: EventTarget) => {
```

Comment:
question: Is this different from using `event.currentTarget.closest('a')`? (ref: https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)

Reply:
oh nice, refactored to use closest https://github.com/hashicorp/web-platform-packages/pull/72/commits/76f2fa4ed39e3f013cd762595cc4a52cd09caebb